rasa/nlu/selectors/response_selector.py

```diff
@@ -273,34 +274,39 @@ def __init__(
                 otherwise initializes it with random weights.
             sparse_feature_sizes: Sizes of the sparse features the model was trained on.
         """
-        component_config = component_config or {}
 
-        # the following properties cannot be adapted for the ResponseSelector
+        # Fill in defaults that are expected by DIET but cannot be specified
+        # for the ResponseSelector
+        component_config = component_config or {}
         component_config[INTENT_CLASSIFICATION] = True
         component_config[ENTITY_RECOGNITION] = False
         component_config[BILOU_FLAG] = None
 
-        # Initialize defaults
-        self.responses = responses or {}
-        self.all_retrieval_intents = all_retrieval_intents or []
-        self.retrieval_intent = None
-        self.use_text_as_label = False
-
         super().__init__(
             component_config,
             index_label_id_mapping,
             entity_tag_specs,
             model,
-            finetune_mode=finetune_mode,
             sparse_feature_sizes=sparse_feature_sizes,
+            finetune_mode=finetune_mode,
         )
 
+        self.retrieval_intent = self.component_config[RETRIEVAL_INTENT]
+        self.use_text_as_label = self.component_config[USE_TEXT_AS_LABEL]
+        self.label_attribute = (
+            RESPONSE if self.use_text_as_label else INTENT_RESPONSE_KEY
+        )
+        self.responses = responses or {}
+        self.all_retrieval_intents = all_retrieval_intents or []
+
     @property
     def label_key(self) -> Text:
+        """Returns the generic `label` key."""
         return LABEL_KEY
 
     @property
     def label_sub_key(self) -> Text:
+        """Returns the generic `id` key."""
```

Comment:
I mean the generic "LABEL_SUB_KEY". I'm not even sure why this function exists. In the DIETClassifier, these functions also return generic keys (because specific keys are copied over to generic keys; cf. discussion in other PR), and changing that to specific keys (i.e. make the functions appear more useful) doesn't work because other modules expect these generic ones.
Any suggestion for a better docstr?

Reply:
Ahh you mean the constant `LABEL_SUB_KEY="id"`.
Btw, it seems to me that you don't need this function inside `ResponseSelector` since you are setting `component_config[INTENT_CLASSIFICATION] = True` because of which the base class (`DIETClassifier`) will always return `LABEL_SUB_KEY`.

Reply:
Let's keep this one in case we refactor the classes one day and change how label_sub_key works (and the copying of keys in the create_data / the keys in create_label)